src/SFA.DAS.Employer.Profiles.Web/Controllers/UserController.cs

```diff
@@ -27,4 +32,48 @@ public IActionResult ChangeSignInDetails()
         var model = new ChangeSignInDetailsViewModel(_configuration["ResourceEnvironmentName"]);
         return View(model);
     }
+
+    [HttpGet]
+    [Route("add-user-details", Name = RouteNames.AddUserDetails)]
+    public IActionResult AddUserDetails()
+    {
+        return View(new AddUserDetailsModel(_configuration["ResourceEnvironmentName"]));
+    }
+
+    [HttpPost]
+    [Route("add-user-details", Name = RouteNames.AddUserDetails)]
+    public async Task<IActionResult> AddUserDetails(AddUserDetailsModel model)
+    {
+        // check if the model state is valid.
+        if (!ModelState.IsValid)
+        {
+            return View(new AddUserDetailsModel(_configuration["ResourceEnvironmentName"])
+            {
+                ErrorDictionary = ModelState
+                    .Where(x => x.Value is {Errors.Count: > 0})
+                    .ToDictionary(
+                        kvp => kvp.Key,
+                        kvp => kvp.Value?.Errors.Select(e => e.ErrorMessage).FirstOrDefault()
+                    ),
+                FirstName = model.FirstName,
+                LastName = model.LastName
+            });
+        }
+
+        // read the claims from the ClaimsPrincipal.
+        var userId = HttpContext.User.Claims.First(c => c.Type.Equals(ClaimTypes.NameIdentifier)).Value;
+        var email = HttpContext.User.Claims.First(c => c.Type.Equals(ClaimTypes.Email)).Value;
+
+        // Add the user details to the repository via Apim.
+        _ = await _accountsService.UpsertUserAccount(userId, new UpsertAccountRequest
+        {
+            FirstName = model.FirstName,
+            Email = email,
+            LastName = model.LastName,
+            GovIdentifier = userId
+        });
+
+        // re-direct the user to the default home page of manage apprenticeship service.
+        return RedirectToRoute(model.RedirectUrl);
```

Comment:
This just wants to be 
```suggestion
        return Redirect(model.RedirectUrl);
```

Reply:
Updated the PR with the recommended changes with more unit testing coverage.